### PR TITLE
Refactor Native Code ID

### PR DESCRIPTION
**Pull request by iamyulong**
## Summary

Replaces native code ID constants with a proper enum, for name consistency and better compiler checks.

### CODE REVIEW

File: radix-engine/src/system/bootstrap.rs [hunk 1]
```diff
@@ -489,7 +489,7 @@ pub fn create_system_bootstrap_flash(
         (
             PACKAGE_PACKAGE,
             PackageNativePackage::definition(),
-            PACKAGE_CODE_ID,
+            NativeCodeId::PackageCode1 as u64 as u64,
```
Comment (0xOmarA) [inline radix-engine/src/system/bootstrap.rs:492]: `as u64 as u64`?

**Review by 0xOmarA — APPROVED** [review]
Looks good!

File: radix-engine/src/system/bootstrap.rs (hunk 1 above)
Comment (iamyulong) [reply radix-engine/src/system/bootstrap.rs:492]: This is accidental. Will update.

**Comment by iamyulong** [comment]
Github is having issue pulling LFS assets, but the determinism test passed for the previous commits. I'm merging ignoring the CI failure.

**Pull request merged**